indico/modules/rb_new/client/js/modules/calendar/Calendar.jsx

```diff
@@ -122,17 +125,25 @@ class Calendar extends React.Component {
 
     onClickLabel = (id) => {
         const {pushState} = this.props;
-        pushState(`/calendar/${id}/details`);
+        pushState(`/calendar/rooms/${id}/details`);
     };
 
     closeModal = () => {
         const {pushState} = this.props;
         pushState('/calendar');
     };
 
+    openDetailsModal = (booking) => {
+        const {pushState} = this.props;
+        pushState(`/calendar/bookings/${booking.id}/details`);
```

Comment:
I think you want to pass `true` as the second argument thus preserving the queryString (I think we will have filters there so opening Modal would probably clear the query parameters)

Reply:
I didn't add it here for now intentionally. Because with no query parameters I end up with 'undefined' in the querystring

Reply:
weren't filters already added there?

Reply:
pushState calls to open modals are going  away anyway, so I wouldn't worry too much about any code related to opening modals